lib/cocoapods-core/podfile/target_definition.rb

```diff
@@ -753,17 +753,18 @@ def parse_configuration_whitelist(name, requirements)
       #         If :subspecs is the only key in the hash, the hash
       #         should be destroyed because it confuses Gem::Dependency.
       #
-      # @return [void]
+      # @return [TrueClass,FalseClass] Returns true if we added new subspecs
       #
       def parse_subspecs(name, requirements)
         options = requirements.last
-        return requirements unless options.is_a?(Hash)
+        return false unless options.is_a?(Hash)
 
         subspecs = options.delete(:subspecs)
         subspecs.each do |ss|
           store_pod("#{name}/#{ss}", *requirements.dup)
         end if subspecs
         requirements.pop if options.empty?
+        !subspecs.nil?
```

Comment:
Not exactly, if it's documented to return a Boolean it should 100% return a boolean.

```ruby
> [].is_a?(TrueClass)
=> false
> [] == true
=> false
```